lib/functional-constraints/matchingKeys.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const _ = require('lodash');
+const jsonschema = require('jsonschema');
+
+const actionTypes = ['triggers', 'searches', 'creates'];
+
+const matchingKeys = definition => {
+  // if none of the actionTypes are in this, we don't have a top-level definition
+  if (!_.some(actionTypes.map(t => definition[t]), Boolean)) {
+    return [];
+  }
```

Comment:
Doesn't looks like this `if` block is necessary. The `for` loop that follows next will safely exit even `definition` doesn't have any of `actionTypes`. Or am I missing something?